url7606378bcc_ljy2010a_protobuf2_tgz-pJ8-com_game_gameObj_bufBuilder_ReflectBuilderJ8/src/com/game/Notice/GameAction.java

```diff
@@ -0,0 +1,34 @@
+package com.game.Notice;
+
+import com.game.protobuf.GameProto;
+
+import java.lang.reflect.InvocationTargetException;
+import java.lang.reflect.Method;
+
+public class GameAction {
+
+	public static void call(GameProto.GameMsg agr_GameMsg)throws IllegalAccessException, IllegalArgumentException,
+            InvocationTargetException, NoSuchMethodException,
+            SecurityException, InstantiationException, ClassNotFoundException{
+		System.out.println("GameAction call");
+		//TODO:  reflect to detail gameObj and func
+        GameProto.GameAction gameAction= agr_GameMsg.getNotice().getGameAction();
+        String actionType = gameAction.getActionType();
+        String objType =    gameAction.getGameObj().getGameObjType();
+
+
+        gameAction.getGameObj().getWarShip().getWarShipLevel();
+
+        Class<?> c = Class.forName("com.game.Notice." + objType);
+
+        Object invokeTester = c.newInstance();
+
+        Method method = c.getMethod(actionType,agr_GameMsg.getClass());
+
+        Object result = method.invoke(invokeTester, agr_GameMsg);
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

Catch `InvocationTargetException` explicitly when you call `Method.invoke()`. Use `Throwable.getCause()` or `getTargetException()` in the catch handler to access more information about the underlying exception. See the following diffs on GitHub that make this code change: [diff1](https://github.com/EsfingeFramework/gamification/commit/20792047e6c69457cca03e877f2ada578188f32a#diff-6808c12fe9ddaea042db84f95dd5ba78adab99d0cb18bb4d85916a8723530768L21), [diff2](https://github.com/cucumber-attic/cuke4duke/commit/f4821d400cf2590126f1298a8dff9f7a33d45dc5#diff-80adaffc20c0e6e7d2e42312386ffe1ddc10213bf5b8e8919895fd85d666c8deL34)

Learn more: [InvocationTargetException in the Java Platform, Standard Edition 8 API Specification](https://docs.oracle.com/javase/8/docs/api/java/lang/reflect/InvocationTargetException.html)